### PR TITLE
feat: support subscribe logs

### PR DESCRIPTION
**Pull request by RetricSu**

### CODE REVIEW

File: packages/api-server/src/db/index.ts [hunk 1]
```diff
@@ -420,6 +420,24 @@ async function filterLogsByTopics(
...
+  logs: Log[],
+  address: HexString | undefined
+): Promise<Log[]> {
+  // match anything
+  if (!address) {
+    return logs;
+  }
+
+  let result: Log[] = [];
+  for await (let log of logs) {
```
Comment (classicalliu) [inline packages/api-server/src/db/index.ts:433]: await ?

File: packages/api-server/src/ws/methods.ts [hunk 2]
```diff
@@ -34,15 +37,38 @@ export function wrapper(ws: any, _req: any) {
...
+          result: result.map((log) => toApiLog(log)),
+          subscription: id,
+        },
+      };
+      ws.send(JSON.stringify(obj));
+    });
+  };
+
   blockEmitter.getEmitter().on("newHeads", blockListener);
+  blockEmitter.getEmitter().on("logs", logsListener);
```
Comment (classicalliu) [inline packages/api-server/src/ws/methods.ts:71]: Release this source when close connection. See line 75 as an example.

File: packages/api-server/src/ws/methods.ts [hunk 3]
```diff
@@ -22,9 +24,10 @@ export function wrapper(ws: any, _req: any) {
     });
   }
 
-  let resultId = 0;
-  const newHeadsIds: Set<number> = new Set();
+  const newHeadsIds: Set<HexNumber> = new Set();
   const syncingIds: Set<HexNumber> = new Set();
+  const logsIds: Set<HexNumber> = new Set();
```
Comment (classicalliu) [inline packages/api-server/src/ws/methods.ts:29]: Seems `logsQueryMaps` is enough ?

**Review by classicalliu — CHANGES_REQUESTED** [review]

File: packages/api-server/src/db/index.ts (hunk 1 above)
Comment (RetricSu) [reply packages/api-server/src/db/index.ts:433]: seems no need to use here

File: packages/api-server/src/ws/methods.ts (hunk 2 above)
Comment (RetricSu) [reply packages/api-server/src/ws/methods.ts:67]: done

**Review by classicalliu — APPROVED** [review]

**Pull request merged**